src/migrateTable.ts

```diff
@@ -65,13 +69,24 @@ export function migrateTable(
 
   const legacyColumns =
     legacyTableState.value?.body?.configuration?.tabular?.columns;
-  const columnWidths = legacyColumns
-    ? _migrateTableColumnWidths({
-        legacyColumns,
-        mapping,
-        cube,
-      })
-    : {};
+
+  const columnsAxis = query.mdx?.axes[0];
```

Comment:
I believe that you are referring to the maximum level depth on rows here.

```suggestion
  const rowsAxis = query.mdx?.axes[0];
```

Also, this will only work if their query has the `ON ROWS` clause before the `ON COLUMNS` one. If some of their queries don't follow this rule, then the migration will yield unexpected results regarding this width.

Instead, we should do something like:

```
query.mdx.axes.find(axis => axis.name === "ROWS");
```
But note that `axis.name` can actually be different things: the case can vary and it can be a number too. See [UI-8726](https://activeviam.atlassian.net/browse/UI-8726) for a recent example of related bug.

Reply:
There is another possible way and it is to derive the level depth from the mapping instead of the query, it might be simpler.